mscore/scoreview.cpp

```diff
@@ -5888,6 +5888,16 @@ void ScoreView::cmdAddRemoveBreaks()
                         }
                   }
             }
+      // deal with last measure of selection, but skip last measure of score
+      if (count == interval && endMeasure != endMeasure->system()->lastMeasure()) {
```

Comment:
Instead of waiting until the end of the loop and basically duplicating the contents of the loop, why not just change the loop end codition?  Seems this code doesn't handle the "lock" case.

I could think of a couple of ways to do this, but maybe the simplest is to change the loop end condition to simply "true", but add "if (m == endMeasure) break;" at the end of the loop.